evalml/tests/automl_tests/test_automl.py

```diff
@@ -2685,96 +2685,132 @@ def test_iterative_algorithm_pipeline_hyperparameters_make_pipeline_other_errors
+                "columns": ["Column_0", "Column_1", "Column_2"]
+            },
+            "Imputer": {"numeric_impute_strategy": "most_frequent"},
+            "Random Forest Classifier": {"n_estimators": 200, "max_depth": 11},
+        }
 
-@patch("evalml.pipelines.BinaryClassificationPipeline.score")
-@patch("evalml.pipelines.BinaryClassificationPipeline.fit")
-def test_iterative_algorithm_pipeline_hyperparameters_make_pipeline(
-    mock_fit, mock_score, X_y_multi
-):
-    X, y = X_y_multi
-    custom_hyperparameters = {"Imputer": {"numeric_impute_strategy": ["mean"]}}
-    larger_custom = {
-        "Imputer": {"numeric_impute_strategy": ["most_frequent", "mean"]},
-        "Extra Trees Classifier": {
-            "max_depth": [4, 5, 6, 7],
-            "max_features": ["auto", "log2"],
-        },
-    }
-    estimators = get_estimators("multiclass", [ModelFamily.EXTRA_TREES])
-    pipelines = [
-        make_pipeline(X, y, estimator, "multiclass", None, custom_hyperparameters)
-        for estimator in estimators
-    ]
+    if pipelines:
+        component_graph_ = [
+            "Drop Columns Transformer",
+            "Imputer",
+            "Random Forest Classifier",
+        ]
+        pipeline_ = [
+            MulticlassClassificationPipeline(
+                component_graph=component_graph_, parameters=pipeline_parameters_
+            )
+        ]
 
-    automl = AutoMLSearch(
-        X_train=X, y_train=y, problem_type="multiclass", allowed_pipelines=pipelines
-    )
-    automl.search()
-    assert automl.best_pipeline.hyperparameters["Imputer"][
-        "numeric_impute_strategy"
-    ] == ["mean"]
+    if automl_parameters:
+        automl_parameters_ = {
+            "Drop Columns Transformer": {
+                "columns": ["Column_0", "Column_1", "Column_2"]
+            },
+            "Random Forest Classifier": {"n_estimators": 201},
+        }
+    if custom_hyperparameters:
+        custom_hyperparameters_ = {
+            "Imputer": {"numeric_impute_strategy": Categorical(["mean"])},
+            "Random Forest Classifier": {
+                "max_depth": Integer(4, 7),
+                "n_estimators": Integer(190, 210),
+            },
+        }
 
-    invalid_pipelines = [
-        make_pipeline(X, y, estimator, "multiclass", None, larger_custom)
-        for estimator in estimators
-    ]
     automl = AutoMLSearch(
         X_train=X,
         y_train=y,
         problem_type="multiclass",
-        allowed_pipelines=invalid_pipelines,
+        allowed_pipelines=pipeline_,
+        pipeline_parameters=automl_parameters_,
+        custom_hyperparameters=custom_hyperparameters_,
+        max_batches=4,
     )
     automl.search()
 
-    assert automl.best_pipeline.hyperparameters["Imputer"][
-        "numeric_impute_strategy"
-    ] == ["most_frequent", "mean"]
+    for i, row in automl.full_rankings.iterrows():
+        if "Random Forest Classifier" in row["pipeline_name"]:
```

Comment:
I see we're only checking RF classifier here. But I'm guessing automl search ran other pipelines. I know we have mocking enabled which is good. But to save some runtime, could you set `model_family` to random forest in this test? That way we won't run other families. Could also turn down `max_batches` to 2.